nucliadb/src/nucliadb/common/maindb/pg.py

```diff
@@ -44,13 +44,32 @@
 );
 """
 
+logger = logging.getLogger(__name__)
 
+# Request Metrics
 pg_observer = metrics.Observer(
     "pg_client",
     labels={"type": ""},
 )
 
-logger = logging.getLogger(__name__)
+# Pool metrics
+POOL_METRICS_COUNTERS = {
+    # Requests for a connection to the pool
+    "requests_num": metrics.Counter("pg_client_pool_requests_total"),
+    "requests_queued": metrics.Counter("pg_client_pool_requests_queued_total"),
+    "requests_errors": metrics.Counter("pg_client_pool_requests_errors_total"),
+    "requests_wait_ms": metrics.Counter("pg_client_pool_requests_queued_seconds_total"),
+    "usage_ms": metrics.Counter("pg_client_pool_requests_usage_seconds_total"),
+    # Pool opening a connection to PG
+    "connections_num": metrics.Counter("pg_client_pool_connections_total"),
+    "connections_ms": metrics.Counter("pg_client_pool_connections_seconds_total"),
+}
+POOL_METRICS_GAUGES = {
+    "pool_size": metrics.Gauge("pg_client_pool_connections_open"),
+    # The two below most likely change too rapidly to be useful in a metric
+    "pool_available": metrics.Gauge("pg_client_pool_connections_available"),
+    "requests_waiting": metrics.Gauge("pg_client_pool_requests_waiting"),
```

Comment:
we could compute the rolling average in memory and report that, but not sure if it's worth the complexity

Reply:
Yeah, but then you have to loop pretty frequently in order to get a value. And you can get an idea of average usage with other metrics (e.g: average queue time). I don't think it's worth the complexity.

I thought about removing them, but they might be useful if things go too badly and a queue starts to build up. Since it's only two metrics, I left them around in case they are useful some day in an emergency.